libcontainer/mount_linux.go

```diff
@@ -229,16 +229,32 @@ func mountFd(nsHandles *userns.Handles, m *configs.Mount) (*mountSource, error)
 		sourceType = mountSourceOpenTree
 
 		// Configure the id mapping.
-		usernsFile, err := nsHandles.Get(userns.Mapping{
-			UIDMappings: m.UIDMappings,
-			GIDMappings: m.GIDMappings,
-		})
-		if err != nil {
-			return nil, fmt.Errorf("failed to create userns for %s id-mapping: %w", m.Source, err)
+		var usernsFile *os.File
+		if m.Mapping.NamespacePath == "" {
+			usernsFile, err = nsHandles.Get(userns.Mapping{
+				UIDMappings: m.Mapping.UIDMappings,
+				GIDMappings: m.Mapping.GIDMappings,
+			})
+			if err != nil {
+				return nil, fmt.Errorf("failed to create userns for %s id-mapping: %w", m.Source, err)
+			}
+		} else {
+			usernsFile, err = os.Open(m.Mapping.NamespacePath)
+			if err != nil {
+				return nil, fmt.Errorf("failed to open existing userns for %s id-mapping: %w", m.Source, err)
+			}
```

Comment:
I think the path joining logic is on this commit by mistake, but it is fine.

Reply:
Which path joining logic?

Reply:
Never mind.

This code is joining an existing namespace with `os.Open()`. However, this is part of the commit `mount: add support for ridmap and idmap`, which is confusing.

I think you might have done this to avoid conflicts and writing the same func again, as here you switch to m.Mapping.NamespacePath. And I agree it is needed. A mention that we are also doing the joining logic now in the commit would have helped me to review, as this is unexpected after reading the commit.

But never mind.

Reply:
The reason this was added here is because only `idmap` and `ridmap` allow you to use the container's user namespace configuration (specifically, the userns path) as the source for the idmap attribute. Before that commit, every mount needed to specify the mappings explicitly.

Reply:
Ohhh, sorry I missed it. Thanks!